### PR TITLE
solution

### PR DESCRIPTION
**Pull request by DolgKreker**

### CODE REVIEW

**Review by asbh25 — APPROVED** [review]

File: src/transformState.js [hunk 1]
```diff
@@ -46,7 +46,31 @@
...
  */
 function transformState(state, transforms) {
-  // write code here
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+  const clear = 'clear';
+
+  for (let i = 0; i < transforms.length; i++) {
+    if (transforms[i].operation === add) {
+      for (const key in transforms[i].properties) {
```
Comment (vkryvytskyy) [inline src/transformState.js:55]:
```suggestion
  for (const transform of ) {
    if (transform.operation === add) {
      for (const key in transforms[i].properties) {
```

File: src/transformState.js [hunk 2]
```diff
@@ -46,7 +46,31 @@
  * @param {Object[]} transforms
  */
 function transformState(state, transforms) {
-  // write code here
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+  const clear = 'clear';
+
+  for (let i = 0; i < transforms.length; i++) {
+    if (transforms[i].operation === add) {
```
Comment (vkryvytskyy) [inline src/transformState.js:54]: consider to use if..else or even better switch..case

**Review by vkryvytskyi — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 3]
```diff
@@ -46,7 +46,31 @@
  * @param {Object[]} transforms
  */
 function transformState(state, transforms) {
-  // write code here
+  const add = 'addProperties';
+  const remove = 'removeProperties';
+
+  for (let i = 0; i < transforms.length; i++) {
```
Comment (vkryvytskyy) [inline src/transformState.js:52]:
```suggestion
  for (const transform of transforms) {
```

File: src/transformState.js [hunk 4]
```diff
@@ -46,7 +46,31 @@
...
+    switch (transforms[i].operation) {
```
Comment (vkryvytskyy) [inline src/transformState.js:53]:
```suggestion
    switch (transforms.operation) {
```

**Review by vkryvytskyi — APPROVED** [review]
fix this

**Pull request closed without merging**